### ISSUE TITLE
Wishlist

### ISSUE THREAD

**Issue by androidacy-user** [Androidacy/MagiskModuleManager#5]
Here is a list of features that have been requested and may or may not be considered.

| Feature | Consideration | Notes |
| --- | --- | --- |
| Share button | Maybe | Will probably be implemented only for Androidacy repo |
| Favorite modules | Yes | In planning phase |
| Backup/Restore modules | No | Improbable due to complications with install scripts |

**Comment by osm0sis** [comment]
Please allow modules to use their own update mechanism again.

**Comment by osm0sis** [comment]
Yeah... so I mean I liked Androidacy being an entry-point to finding my modules, but then I want my modules to be allowed to update themselves, as I have intended, and as the app used to work. 🫤

**Comment by pndwal** [comment]
I understand the issue of overseeing module repos... The problems are really why John sunsetted the official original...

John said he hoped to establish a suitable body to oversee official repos again at a later time but this hasn't materialised...

Meanwhile, proper issues were opened for an interim solution for pushing updates... I made an initial suggestion about using a Developer set JSON for updates and limited documentation... That was debated but was listened to, and it's what we now have officially...

Personally, I will avoid using solutions that override/bypass supported/intended functions including supported update JSON... Altering any code seems dubious to me, but should at least be done with developer permission and agreement and not replace/dispense with Magisk utility... Doing so otherwise will only alienate developers and users...

The onus for vetting modules for suitability/security in the absence of official oversight now clearly rests with end users/modders, including available updates, but you can police those you host of course, and updates as well...

In the case of 'bad apples', malware will of course be pushed from your repo(s) if you don't catch and remove  modules... But you have access to any update JSON being called from modules you host also, so why not simply extend your solution to download and scan updates as soon as a dev pushes changes to JSON? Simple!

If you don't want it to seem "like they're using this 'bad apples' argument to force all downloads to come from their website, and therefore serve more ads", you should listen to this argument.

**Comment by peggy-48** [comment]
Why don't you make back fox mm and let it at its place and why you put tons of ads on ur website coz github pages exist anyways so no subscrip. so just why not give main dev's github repo urls too for magisk mod

**Comment by Masamune3210** [comment]
I don't like it either, but threatening is not the solution. Just use one of the other ways to do this or other apps that offer the same abilities.
